app/views/unattended/provisioning_templates/snippet/subscription_manager_setup.erb

```diff
@@ -0,0 +1,144 @@
+<%#
+kind: snippet
+name: subscription_manager_setup
+model: ProvisioningTemplate
+snippet: true
+description: |
+  This snippet installs and configures subscription-manager that is required for host registration.
+  The snippet is included in the "global_registration" template and in the "redhat_register" snippet.
+  The latter ensures host registration during the provisioning process.
+-%>
+
+# Select package manager for the OS (sets the $PKG_MANAGER* variables)
+if [ -z "$PKG_MANAGER" ]; then
+ <%= snippet 'pkg_manager' -%>
+fi
+
+# Define the path to rhsm.conf
+RHSM_CFG=/etc/rhsm/rhsm.conf
+
+<% if @subman_setup_scenario == 'registration' -%>
+  # Backup the original rhsm.conf file
+  if [ -f $RHSM_CFG ] ; then
+    test -f $RHSM_CFG.bak || cp $RHSM_CFG $RHSM_CFG.bak
+  fi
+<% end -%>
+
+<% if plugin_present?('katello') -%>
+  # Define the path to the Katello server CA certificate
+  KATELLO_SERVER_CA_CERT=/etc/rhsm/ca/katello-server-ca.pem
+
+  # If SSL_CA_CERT is not set, create a temporary file for it
+  if [ -z "$SSL_CA_CERT" ]; then
+    SSL_CA_CERT=$(mktemp)
+    cat << EOF > "$SSL_CA_CERT"
+<%= foreman_server_ca_cert %>
+EOF
+  fi
+
+  <% if @subman_setup_scenario == 'registration' -%>
+    # rhn-client-tools conflicts with subscription-manager package
+    # since rhn tools replaces subscription-manager, we need to explicitly
+    # install subscription-manager after the rhn tools cleanup
+    if [ x$ID = xol ]; then
+      $PKG_MANAGER_REMOVE rhn-client-tools
+      $PKG_MANAGER_INSTALL --setopt=obsoletes=0 subscription-manager
+    fi
+
+    <% if truthy?(@force) -%>
+      # Unregister host and remove all local system and subscription data
+      if [ -x "$(command -v subscription-manager)" ] ; then
+        subscription-manager unregister || true
+        subscription-manager clean
+      fi
+
+      $PKG_MANAGER_REMOVE katello-ca-consumer\*
+    <% end -%>
+  <% end -%>
+
+  # Prepare the SSL certificate
+  mkdir -p /etc/rhsm/ca
+  cp -f $SSL_CA_CERT $KATELLO_SERVER_CA_CERT
+  chmod 644 $KATELLO_SERVER_CA_CERT
+<% end -%>
+
+# Prepare subscription-manager
```

Comment:
@stejskalleos should we check if the machine is RHEL compatible?
For example, [here](https://github.com/theforeman/foreman/blob/develop/app/views/unattended/provisioning_templates/finish/kickstart_default_finish.erb#L58) we are calling the `redhat_register` snipped without verifying the machine's RHEL compatibility.

Reply:
No need, kickstart is specifically for RHEL, if someone is going to use the snippet somewhere else, I expect that the person knows what is doing.